File: app/templates/server_go/api/todo/controller/todocontroller.go
package todocontroller

import (
	"encoding/json"
	"github.com/<%= username %>/<%= appName %>/server/api/todo/dao"
	todo "github.com/<%= username %>/<%= appName %>/server/api/todo/model"
	"github.com/labstack/echo"
	"io/ioutil"
	"net/http"
)

func GetAll(c echo.Context) error {
	ts, _ := tododao.All()

	return c.JSON(http.StatusOK, ts)
}

func NewTodo(c echo.Context) error {
	t := todo.Todo{}

	nt, _ := tododao.NewTodo(t)

	return c.JSON(http.StatusOK, nt)
}

func RemoveTodo(c echo.Context) error {
	id := c.Param("id")

	tododao.DeleteTodo(id)

	return c.String(http.StatusOK, "")
}

File: app/templates/client_ng1/dev/todo/controllers/todo-controller.js
;(function(ng) {
  'use strict';

  ng.module('<%= appName %>')
    .controller('TodoController', ['$log', 'Todo', 'TodoDAO', function($log, Todo, TodoDAO) {
      var self = this;

      self.todo = new Todo();
      self.todos = [];

      self.createTodo = function(todo) {
        TodoDAO
          .createTodo(todo)
          .then(function(newTodo) {
            self.todos.push(newTodo);
            self.todo = new Todo();
          })
          .catch($log.error);
      };

      self.deleteTodo = function(id) {
        TodoDAO
          .deleteTodo(id)
          .then(function() {
            return _getAll();
          })
          .catch($log.error);
      }

      var _getAll = function() {
        return TodoDAO
          .getAll()
          .then(function(todos) {
            return self.todos = todos;
          })
          .catch($log.error);
      }

      _getAll();
    }]);
}(window.angular));
